DOCSP-48386 Add monitoring page

stephmarie17 opened the pull request:
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-48386>
Staging - <https://deploy-preview-1046--docs-node.netlify.app/monitoring-and-logging/monitoring/>

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

rachel-mack commented on source/monitoring-and-logging/monitoring.txt:
```````````````````````````diff
@@ -4,20 +4,730 @@
+Event Subscription Example
+``````````````````````````
+
+You can access one or more SDAM events using the driver by subscribing to them
+in your application. The following example demonstrates connecting to a
+replica set and subscribing to one of the SDAM events created by the MongoDB
+deployment:
```````````````````````````
Our style guide wants us to say "by using" with should make the first sentence "by using the driver by subscribing" :P

You could potentially make the eg steps into a list, but I think this is clear enough. I just think we should highlight that we're formatting the event before logging it. I'm not sure whether we should be doing this in all the examples, or if this is actually a difference in how this event type is returned? Maybe it's not necessary?

```suggestion
You can access one or more SDAM events by subscribing to them
in your application. The following example demonstrates connecting to a
replica set, subscribing to an SDAM event and logging the event in JSON format:
```

rachel-mack commented on source/monitoring-and-logging/monitoring.txt:
```diff
@@ -4,20 +4,730 @@
+Server Discovery and Monitoring Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} creates topology events, also known as SDAM events, when
+there is a change in the state of the instance or cluster that you connected to.
+For example, the driver creates an event when you establish a new connection or
+if the cluster elects a new primary.
```
You could include a link to the [Replication server page](https://www.mongodb.com/docs/manual/replication/) for more information about topology.

```suggestion
The {+driver-short+} creates topology events, also known as SDAM events, when
there is a change in the state of the instance or cluster that you connected to.
For example, the driver creates an event when you establish a new connection or
if the cluster elects a new primary node.
```

rachel-mack commented on source/monitoring-and-logging/monitoring.txt:
```diff
@@ -4,20 +4,730 @@
+     - Created when a connection is checked back into the pool after an operation
+       is executed.
+
+   * - ``connectionPoolCleared``
+     - Created when a connection pool is cleared.
```
What does cleared mean? No more connections available (all connections are checked out) or all connections closed?

stephmarie17 replied:
I think it's all connections closed. I can add to clarify and get tech review.

rachel-mack reviewed (approved):
Some minor stuff and a couple questions:

stephmarie17 commented:
@alexbevi when you have a moment could you take a look at this PR? No code changes, just moved content to align with the recent Java Sync and PyMongo updates. There was one question that came up in review so I updated some content slightly (see above comment). Or if you notice anything out of date/missing that would be helpful to note. Thank you!
